libraries/chain/fork_database.cpp

```diff
@@ -412,6 +410,7 @@ namespace eosio::chain {
    fork_database_t<bsp>::branch_type
    fork_database_impl<bsp>::fetch_branch_impl(const block_id_type& h, uint32_t trim_after_block_num) const {
       branch_type result;
+      result.reserve(8);
```

Comment:
`8` seems like an odd choice if you are going to use non-default.
Seems like we could use `index.size()+1` assuming our total fork count would never be too large.
Same comment for the reserve below.

Reply:
I don't think the size rreally matters, as long as it is big enough for 99% of the cases.
Realistically it makes no difference in speed wether we allocate space for 5, 8, or 10 pointers, it is still one heap allocation. So I just picked 8 as I thought it would be large enough. It is just a reserve so it is not critical.

We could use `index.size()` but I'm not sure if that is just a lookup of a member variable, and even then it is probably slower than just hardcoding a size.

Reply:
Looks like `index.size()` does return a cached size value.

Reply:
OK I'll use that.